core/config/yaml_test.go

```diff
@@ -120,3 +121,24 @@ func TestYamlNode_Callbacks(t *testing.T) {
 	assert.NoError(t, p.RegisterChangeCallback("test", nil))
 	assert.NoError(t, p.UnregisterChangeCallback("token"))
 }
+
+type emptystruct struct {
+	Slice []string
+}
+
+var emptyyaml = []byte(`
+emptystruct:
+  nonexist: true
+`)
+
+func TestPopulateMismatchedStruct(t *testing.T) {
+	provider := NewProviderGroup("global", NewYAMLProviderFromBytes(emptyyaml))
+	es := emptystruct{}
+	empty := reflect.New(reflect.TypeOf(es)).Elem().Interface()
+	assert.True(t, reflect.DeepEqual(empty, es))
+
+	provider = NewProviderGroup("global", NewYAMLProviderFromBytes([]byte(``)))
```

Comment:
This should be two different tests, since they're testing different things, right?